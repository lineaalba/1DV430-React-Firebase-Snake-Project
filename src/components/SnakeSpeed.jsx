/**
 * The SnakeSpeed component.
 * 
 * @author Filippa Jakobsson
 * @version 1.0
 */

import React from 'react'
import Button from 'react-bootstrap/Button'

class SnakeSpeed extends React.Component {

    /**
    * Callback to send data with by the player chosen speed.
    * @returns chosen level.
    */
    sendData = () => {
        const speed = document.querySelectorAll('.speed')

        for (let i = 0; i < speed.length; i++) {
            let level
            speed[i].addEventListener('click', e => {
                switch (speed[i].textContent) {
                    case 'Easy':
                        level = 120
                        break
                    case 'Moderate':
                        level = 80
                        break
                    case 'Hard':
                        level = 60
                        break
                    case 'Insane':
                        level = 40
                        break
                    default:
                        level = 120
                }
                return this.props.speed(level)
            })
        }
    }
    
    render() {
        return (
            <div className="dropdown">
                <Button data-testid="snakespeed" className="change__speed" variant="outline-dark">Change speed</Button>
                <div className="dropdown__content">
                    <p onClick={this.sendData} className="speed">Easy</p>
                    <p onClick={this.sendData} className="speed">Moderate</p>
                    <p onClick={this.sendData} className="speed">Hard</p>
                    <p onClick={this.sendData} className="speed">Insane</p>
                </div>
            </div>
        )
    }
}

export default SnakeSpeed