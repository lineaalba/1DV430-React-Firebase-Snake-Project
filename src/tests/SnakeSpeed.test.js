import React from 'react'
import { shallow } from 'enzyme'
import SnakeSpeed from '../components/SnakeSpeed'

describe('Tests the SnakeSpeed component', () => {

  it('matches snapshot', () => {
    const wrapper = shallow(<SnakeSpeed />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
  