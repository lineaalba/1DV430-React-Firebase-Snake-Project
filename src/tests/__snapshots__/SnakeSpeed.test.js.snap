// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the SnakeSpeed component matches snapshot 1`] = `"<div class=\\"dropdown\\"><button data-testid=\\"snakespeed\\" type=\\"button\\" class=\\"change__speed btn btn-outline-dark\\">Change speed</button><div class=\\"dropdown__content\\"><p class=\\"speed\\">Easy</p><p class=\\"speed\\">Moderate</p><p class=\\"speed\\">Hard</p><p class=\\"speed\\">Insane</p></div></div>"`;
